visualization/app/codeCharta/util/streetLayoutGenerator.ts

```diff
@@ -0,0 +1,105 @@
+import { CodeMapNode, MetricData, Node, State, BlacklistType, TreeMapAlgorithm, LayoutAlgorithm } from "../codeCharta.model"
+import BoundingBox from "./algorithm/streetLayout/boundingBox"
+import VerticalStreet from "./algorithm/streetLayout/verticalStreet"
+import HorizontalStreet from "./algorithm/streetLayout/horizontalStreet"
+import House from "./algorithm/streetLayout/house"
+import Rectangle from "./algorithm/rectangle"
+import Point from "./algorithm/point"
+import { CodeMapHelper } from "./codeMapHelper"
+import TreeMap from "./algorithm/treeMap/treeMap"
+import SquarifiedTreeMap from "./algorithm/treeMap/squarifiedTreeMap"
+import { StreetOrientation } from "./algorithm/streetLayout/street"
+import { LayoutHelper, LayoutNode } from "./layoutHelper"
+
+export interface StreetLayoutValuedCodeMapNode {
+	data: CodeMapNode
+	value: number
+	rect: Rectangle
+	zOffset: number
+}
+
+export class StreetLayoutGenerator {
+	private static MARGIN_SCALING_FACTOR = 0.02
+
+	public static createStreetLayoutNodes(map: CodeMapNode, state: State, metricData: MetricData[]): Node[] {
+		const isDeltaState = state.files.isDeltaState()
+		const metricName = state.dynamicSettings.areaMetric
+		const mergedMap = LayoutHelper.mergeDirectories(map, metricName)
+		const maxTreeMapFiles = state.appSettings.maxTreeMapFiles
+		const childBoxes = this.createBoxes(mergedMap, metricName, state, StreetOrientation.Vertical, 0, maxTreeMapFiles)
+		const rootStreet = new HorizontalStreet(mergedMap, childBoxes, 0)
+		rootStreet.calculateDimension(metricName)
+		const margin = state.dynamicSettings.margin * StreetLayoutGenerator.MARGIN_SCALING_FACTOR
+		const layoutNodes: StreetLayoutValuedCodeMapNode[] = rootStreet.layout(new Point(0, 0), margin)
+		const maxHeight = metricData.find(x => x.name == state.dynamicSettings.heightMetric).maxValue
+		const heightScale = 1 //TODO: apply correct height after scaling layout size
+		return layoutNodes.map(streetLayoutNode => {
+			return LayoutHelper.buildNodeFrom(streetLayoutNode as LayoutNode, heightScale, maxHeight, state, isDeltaState)
+		})
+	}
+
+	private static createBoxes(
```

Comment:
Can you refactor that method to be shorter and/or split into multiple more readable methods?